### PR TITLE
fix: make CoW great again

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR uses the https://github.com/cowprotocol/app-data package to generate proper CoW AppData, in an effort to fix CoW trades (probably failing because of slippage), as well as making sure we are up-to-date on the CoW AppData version and are able to use the additional metadata fields programatically in the future.

This is as opposed to CoW's previously provided appdata (https://github.com/shapeshift/lib/pull/956) which referenced a static IPFS CID with no metadata.

As for why we're doing this weird `appData` / `appDataHash` dance, this is by design with CoW, see:

https://api.cow.fi/docs/#/default/post_api_v1_quote
https://github.com/cowprotocol/cowswap/blob/a11703f4e93df0247c09d96afa93e13669a3c244/apps/cowswap-frontend/src/legacy/utils/trade.ts#L236

tl;dr here is
- CoW *endpoints* accept either `appData` as an hex string (given there's a matching IPFS document ID, which there was for the previous CoW-provided AppData), or as a JSON-stringified, *stable-ordered* string, in which case we need to pass the matching keccac256 for it as `appDataHex`. If we do so, there is no need for the document to exist in IPFS
- CoW *signature* requires `appData` as an hex string, and doesn't accept an additional `appDataHash`

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

- Partner app code might be broken, ensure you can still see it in the metadata in CoW explorer under "AppData" (see screenshots)

## Testing

- CoW swaps complete
- CoW swaps requests/responses still have an `appData` field
- CoW swaps requests/responses now have an `appDataHash` field
- In CoW explorer, expanding the `AppData` displays a JSON containing our app code (`shapeshift`) as well as additional order type (`market`) and slippage bips meta (`50`)

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5458** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5458" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5458?utm_source=stack-comment).

**Comment by woodenfurniture** [comment]
Cow trades working after tweaking the webpack config slightly. 

The webpack changes were nested under `isProduction` which is why it wasnt working for local build. One of the changes doesnt seem to be required :) 

https://explorer.cow.fi/orders/0xd288f5a4c31ad125aceba048540969d9024ab50f9480755d500996da056cb17734b5d5153fc1a2107a311ce6412fb69a30d78e156528a7dd?tab=overview

image
image
image
image

File: package.json [hunk 1]
```diff
@@ -78,6 +78,7 @@
     "@chakra-ui/react": "^2.8.0",
     "@chakra-ui/system": "^2.6.0",
     "@chakra-ui/tag": "^3.1.0",
+    "@cowprotocol/app-data": "^1.1.0-RC.0",
```
Comment (gomesalexandre) [inline package.json:81]: This is weird, but this is what is actually published as stable 🤷🏽

**Review by woodenfurniture — APPROVED** [review]
